src/middleware/authMiddleware.ts

```diff
@@ -5,33 +5,114 @@ import { JwtPayload } from "../types/auth";
 import { User } from "../models/User";
 import { AppDataSource } from "../config/db";
 
-declare global {
-  namespace Express {
-    interface Request {
-      user?: JwtPayload;
-    }
-  }
-}
-
-export const authenticateToken = (
+export const authenticateToken = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
-  const authHeader = req.headers.authorization;
-  const token = authHeader && authHeader.split(" ")[1];
+
+  const authHeader = 
+    req.headers['authorization'] || 
+    req.headers['Authorization'] || 
+    req.headers['AUTHORIZATION'];
+
+  console.log(`Raw Authorization Header: ${authHeader}`);
+
+  if (!authHeader) {
+    console.error('No authorization header found');
+    return res.status(401).json({ 
+      message: "Unauthorized: No authorization header",
+      details: "Authorization header is missing" 
+    });
+  }
+
+  const normalizedHeader = authHeader.toString().trim();
+  
+  console.log(`Normalized Authorization Header: ${normalizedHeader}`);
+
+  const tokenMatch = normalizedHeader.match(/^Bearer\s+(.+)$/i);
```

Comment:
IMHO; I thing this is too much work as regex takes a lot of time to execute. 
I would have done this  `if (authHeader && authHeader.startsWith('Bearer ')) {return authHeader.split(' ')[1];}`
Like I said in my opinion. it does not mean you are wrong

Reply:
I initially used that method, but I encountered some errors with the "Authorization" header for the frontend. While debugging, I decided to try a regex approach to see if it would resolve the issue. However, I noted that...

Noted and thanks...